app/mailers/user_mailer.rb

```diff
@@ -46,17 +44,82 @@ def work_collaborator(user, work)
     mail to: @user.email, subject: "You've been added to #{@work.title}"
   end
 
-  def nightly_user_activity(user)
-    @user = user
-    user_activity(@user)
-    if @active_user
-      mail to: @user.email, subject: "New FromThePage Activity"
+  def nightly_user_activity(user_activity)
+    @user_activity = user_activity
+
+    if @user_activity.has_contributions?
+      mail to: @user_activity.user.email, subject: "New FromThePage Activity"
     end
   end
 
   private
+
   def add_inline_attachments!
     attachments.inline["logo.png"] = File.read("#{Rails.root}/app/assets/images/logo.png")
   end
 
+  class Activity
+    attr_accessor :user, :added_works, :active_pages, :active_translations, :active_note_pages
+
+    def initialize(user:, added_works:, active_pages:, active_translations:, active_note_pages:)
+      @user = user
+      @added_works = added_works
+      @active_pages = active_pages
+      @active_translations = active_translations
+      @active_note_pages = active_note_pages
+    end
+
+    class << self
+      def build(user)
+        #find which pages the user has worked on
+        user_page_ids ||= user.deeds.pluck(:page_id).uniq.compact
+
+        Activity.new(
+          {
+            user: user,
+            added_works: works_added_to_users_collection_in_past_day(user),
+            active_pages: user_pages_edited_in_past_day(user, user_page_ids),
```

Comment:
I've moved all of the query logic into clearly-named methods, so we have a better idea of what they're providing. The goal is make editing the components of the `Activity` object easier too.